_posts/2020-11-2-kubernetes-webhooks-made-easy-with-openfaas-part-2.md

```diff
@@ -0,0 +1,291 @@
+---
+title: "Kubernetes Webhooks made easy with OpenFaaS - Part 2"
+description: "In this post you'll learn how to write Kubernetes Admission webhooks using OpenFaaS functions - Part 2"
+date: 2020-11-2
+image: /images/2020-10-27-k8s-validatingwebhook-openfaas/puzzle.jpg
+categories:
+  - arkade
+  - kubectl
+  - faas-cli
+  - admissionwebhooks
+  - mutatingadmissionwebhooks
+  - k8s extensibility
+author_staff_member: developer-guy
+dark_background: true
+---
+
+In this post you'll learn how to write Kubernetes Admission webhooks using OpenFaaS functions - Part 2
+
+<p align="center">
+<img height="128" src="/images/openfaas/kubernetes.png">
+</p>
+
+## Introduction to Kubernetes Mutating Admission webhooks
+In the previous [part](https://www.openfaas.com/blog/kubernetes-webhooks-made-easy-with-openfaas/), we talked about what Kubernetes Admission Webhooks are and how can we use Validating Admission Webhook.
+
+Today we are gonna talk about Mutating Admission Webhook and demonstrate how can we use it as OpenFaaS Function.
+
+The main difference between the two types of admission webhook are pretty self-explanatory: validating webhooks can reject a request, but they cannot modify the object they are receiving in the admission request, while mutating webhooks can modify objects by creating a patch that will be sent back in the admission response and can reject a request too. If a webhook rejects a request, an error is returned to the end-user.
```

Comment:
Remove `self-explanatory`. This may be obvious to you, but consider how the reader may feel reading this.

Reply:
Thank you, done